### ISSUE TITLE
Does INSTALL_REFERRER receiver still work on Android Oreo?

### ISSUE THREAD

**Issue by jozuasijsling** [adjust/android_sdk#295]
Hi there,

We're upgrading our target and compile SDK versions to Oreo, and we're curious how Adjust deals with the new [limits on broadcast receivers](https://developer.android.com/about/versions/oreo/background.html#broadcasts). Since `INSTALL_REFERRER` is not part of the announced [exception list](https://developer.android.com/guide/components/broadcast-exceptions.html), we're worried that the receiver won't work anymore. Here's the relevant manifest snippet.

```xml
<receiver
  android:name="com.adjust.sdk.AdjustReferrerReceiver"
  android:exported="true">
  <intent-filter>
    <action android:name="com.android.vending.INSTALL_REFERRER" />
  </intent-filter>
</receiver>
```

**Comment by jozuasijsling** [comment]
My guess is that Android sends this intent only to the package that was installed. If so, it's an explicit intent and it should not be affected by the new limitations.

**Comment by uerceg** [comment]
Hi @jozuasijsling,

Correct. `INSTALL_REFERRER` is something that in this Google's definition (explicit vs. implicit) would be named explicit, since it's app specific. Also, this intent is not fired by system, but rather Google Play Services library.

We have tested `AdjustReferrerReceiver` functionality in capturing `INSTALL_REFERRER` intent that we have fired to device running Android O and nothing changed functionality-wise when it comes to that topic.

We're keeping an eye on this topic and in case something changes, our README will get updated accordingly. So far, all good.

**Comment by jozuasijsling** [comment]
Great to hear and thanks for the quick response! 👍

**Comment by fvdcx** [comment]
Hi, guys @jozuasijsling  @uerceg  any official announcement/doc from Google that says `INSTALL_REFERRER ` broadcast receiver still works well on apps targeting API 26?

**Comment by uerceg** [comment]
Hi @fvdcx 

Sorry for a bit delayed answer. Not that we know of, maybe there is somewhere. Never the less, we are still seeing referrer information incoming to our end from devices with API level 26 and higher, so it definitely works. Google will eventually deprecate old referrer mechanism with`INSTALL_REFERRER` intent as a mean of delivery of referrer information into the app, but from what we know, it still unclear when will this happen. At this moment, this old way and new install referrer API work in parallel and will continue to work until Google decides that they gave enough time to people to migrate to new API.

**Comment by dilip12595** [comment]
can i get complete  sample of INSTALL_REFERRER?? i am implement this but not getting referrer code pls.

**Comment by uerceg** [comment]
How are you testing it?

**Comment by talhakaya** [comment]
Hello, we received a mail from Google saying INSTALL_REFERRER intent will be deprecated in 1st of March 2020. I am not sure how to get rid of it though. As I understand Adjust can work perfectly fine without this intent, so should I just go on and delete it from our AndroidManifest and call it done? Or is there any guide to help us move from the old way to the new way, which is Play Install Referrer API? Thanks.

Asking because we have an already released app that has INSTALL_REFERRER intent in its manifest, and we need to remove it and submit a new version.

**Comment by uerceg** [comment]
👋 @talhakaya 

You can probably find all the answers in [this one](https://github.com/adjust/android_sdk/issues/410).

In case you have any additional questions, feel free to ask.

Cheers

**Comment by talhakaya** [comment]
Thank you for the quick reply, haven't seen that thread, so thanks a lot for the link. It seems to have everything I need :)
